src/rdkafka_conf.c

```diff
@@ -3457,6 +3457,12 @@ int rd_kafka_conf_warn (rd_kafka_t *rk) {
                                      rk->rk_conf.socket_timeout_ms);
         }
 
+        if (rd_kafka_conf_is_modified(rk->rk_conf, "max.poll.interval.ms") &&
+            rd_kafka_conf_is_modified(rk->rk_conf, "group.instance.id"))
```

Comment:
This is not symmetrical to how group.instance.id is used: what happens if group.instance.id is set to the empty string (e.g., disabled)?

You should check if group.instance.id is non-empty here,.